Source/Scene/Model.js

```diff
@@ -1454,6 +1454,191 @@ define([
         return getStringFromTypedArray(loadResources.getBuffer(bufferView));
     }
 
+    function shallowClone(obj) {
+        var newObj = {};
+        for (var key in obj) {
+            if (obj.hasOwnProperty(key)) {
+                newObj[key] = obj[key];
+            }
+        }
+        return newObj;
+    }
+
+    function replaceAllButFirstInString(string, find, replace) {
+        return string.replace(new RegExp(find, 'g'), function(match, offset, all) {
+            return all.indexOf(find) === offset ? match : replace;
+        });
+    }
+
+    function scaleFromMatrix5Array(matrix) {
+        return [matrix[0], matrix[1], matrix[2], matrix[3],
+                matrix[5], matrix[6], matrix[7], matrix[8],
+                matrix[10], matrix[11], matrix[12], matrix[13],
+                matrix[15], matrix[16], matrix[17], matrix[18]];
+    }
+
+    function translateFromMatrix5Array(matrix) {
+        return [matrix[20], matrix[21], matrix[22], matrix[23]];
+    }
+
+    function stringifyArray(array) {
+        return array.map(function(val) {
+            return val.toFixed(20);
+        });
+    }
+    
+    function modifyShaderForQuantizedAttributes(shader, programName, model, context) {
```

Comment:
It doesn't enter the function if the model doesn't have `WEB3D_quantized_attributes` listed in its extensionsUsed, do we want something more involved than that?